corehq/apps/users/models.py

```diff
@@ -230,6 +230,11 @@ class UserRole(Document):
     permissions = SchemaProperty(Permissions)
     is_archived = BooleanProperty(default=False)
 
+    @classmethod
+    @quickcache(['doc_id'], timeout=0)
```

Comment:
I'm not sure I understand this set of params. Since you're defining the cache on the class it's still going to have a `memoized` lifecycle that spans multiple requests. If you wanted that to be true, then I would think you would want to use the real cache. If you didn't want that to be true then I don't think this works. This may be the source of test failures.